deployer/src/handlers/mod.rs

```diff
@@ -320,6 +333,60 @@ pub async fn get_service_resources(
     }
 }
 
+#[instrument(skip_all, fields(%project_name, %service_name, %resource_type))]
+#[utoipa::path(
+    delete,
+    path = "/projects/{project_name}/services/{service_name}/resources/{resource_type}",
+    responses(
+        (status = 200, description = "Deletes a resource owned by a service."),
+        (status = 500, description = "Database error.", body = String),
+        (status = 404, description = "Record could not be found.", body = String),
+    ),
+    params(
+        ("project_name" = String, Path, description = "Name of the project that owns the service."),
+        ("service_name" = String, Path, description = "Name of the service."),
+        ("resource_type" = String, Path, description = "Resource type.")
+    )
+)]
+pub async fn delete_service_resource(
+    Extension(mut persistence): Extension<Persistence>,
+    // Extension(mut resource_manager): Extension<dyn ResourceManager>,
+    Extension(claim): Extension<Claim>,
+    Path((project_name, service_name, resource_type)): Path<(String, String, String)>,
+) -> Result<()> {
+    let service = persistence
+        .get_service_by_name(&service_name)
+        .await?
+        .ok_or_else(|| Error::NotFound("service not found".to_string()))?;
+
+    let resource_type: shuttle_common::resource::Type =
+        resource_type
+            .parse()
+            .map_err(|e: resource::ParseError| Error::Convert {
+                from: "String".to_string(),
+                to: "ResourceType".to_string(),
+                message: e.to_string(),
+            })?;
+
+    let persistence_resource_type = resource_type.into();
+
+    persistence
+        .get_resource(&service.id, persistence_resource_type, claim)
+        .await?
+        .ok_or_else(|| Error::NotFound("resource not found".to_string()))?;
+
+    // resource_recorder_client
+    // resource_manager
+    //     .delete_resource(&project_name, &resource_type)
+    //     .await
+    //     .map_err(|e| Error::Custom(e.into()))?;
+    persistence
+        .delete_resource(&service.id, persistence_resource_type, claim)
+        .await?;
```

Comment:
I'm thinking we can try to roll the provisioner resource deletion in `persistence.delete_resource`.

Reply:
I think this should be fixed here: https://github.com/shuttle-hq/shuttle/pull/1256/commits/61eb0ebf6b1914f55ecc97e80a4676824ded85bf. LMK if you meant something else.